src/main/kotlin/com/criptext/mail/db/ComposerLocalDB.kt

```diff
@@ -22,13 +22,24 @@ class ComposerLocalDB(val contactDao: ContactDao, val emailDao: EmailDao, val fi
         val files = fileDao.getAttachmentsFromEmail(id)
         val fileKey = fileKeyDao.getAttachmentKeyFromEmail(id)
 
+        val fromContact = if(EmailAddressUtils.checkIfOnlyHasEmail(email.fromAddress)){
+            contactDao.getContact(
```

Comment:
Doing another query to DB is not necessary.  You have the object `contactsFROM[0]` just return that